skcosmo/metrics/_reconstruction_measures.py

```diff
@@ -0,0 +1,651 @@
+from skcosmo.model_selection import train_test_split
```

Comment:
There's a lot of repeat code in these functions, and it makes for an enormous file. Would it be possible to simplify things by creating a pointwise/global flag in the function calls of `reconstruction_distortion`, `reconstruction_error`, etc. that returns the corresponding function version?

Reply:
Can you please answer this question? It is not resolved.

Reply:
I also thought about this but I feel like this makes things less transparent for the user. Under the principle that all user accessible functions should be fully documented. It would probably look a bit like
```
#  public function with documentation for all inputs and for distortion and error
def global_reconstruction(..., type):
    ...
    if type == "error":
        _global_reconstruction_error(...)
    elif type == "distortion":
        _global_reconstruction_distortion(...)

def _global_reconstruction_error(...):
def _global_reconstruction_distortion(...):
```
Now the documentation for distortion and error is entangled in one function. Also the user has an additional input parameter. To avoid this, one would need a public `global_reconstruction_error` and `global_reconstruction_distortion`, but then these also need a documentation because they are accessible by the user.
From all solutions I tend to the repetitive documentation, maybe because sklearn also has a lot of repetitive documentation and this seems to be more sklearn-style for me.